src/to_device.jl

```diff
@@ -0,0 +1,60 @@
+import Adapt
+import ClimaComms
+
+"""
+    out = to_device(device, x::Union{
+        DataLayouts.DataLayout,
+        Spaces.AbstractSpace,
+        Fields.Field,
+        Fields.FieldVector,
+    })
+
+Move `x` backing data to the given `device`.
```

Comment:
```suggestion
Move `x` to the given `device`.
```
sounds simpler to me.